### PR TITLE
Tailwind migration

### PR DESCRIPTION
**Pull request by dolcalmi**
**Implemented**
- Remove bootstrap
- Add tailwind
- Refactor current functionality

**This is a large PR so I will split the next:**
- Transactions screen
- Migrate tailwind to npm (to allow css size reduction and enable config) + Responsive (menu)

**Screenshots**

Screenshot from 2021-10-14 04-21-17
Screenshot from 2021-10-14 04-21-28
Screenshot from 2021-10-14 04-22-16
Screenshot from 2021-10-14 04-24-10
Screenshot from 2021-10-14 04-24-26

### CODE REVIEW

File: src/components/AddToMap.js [hunk 1]
```diff
@@ -20,19 +16,29 @@ const BUSINESS_UPDATE_MAP_INFO = gql`
   }
 `
 
-function AddToMap() {
-  const [businessUpdateMapInfo] = useMutation(BUSINESS_UPDATE_MAP_INFO)
+function useStateWithDep(defaultValue) {
```
Comment (RonanJackson) [inline src/components/AddToMap.js:19]: I'd like to understand this a bit more. Why are we using a custom hook here for the states for `titleVal`, `latitudeVal` etc?

File: src/components/AddToMap.js [hunk 2]
```diff
@@ -52,36 +58,78 @@ function AddToMap() {
...
+          required
+          type="text"
+          placeholder="Enter longitude"
+          value={latitudeVal}
+          onChange={(e) => setLatitudeVal(e.target.value)}
+          disabled={!!disabledClass}
+          className={`${disabledClass} mt-4 shadow appearance-none border rounded w-full py-2 px-3 text-gray-600 focus:outline-none focus:shadow-outline`}
+        />
+      </div>
+      <div className="">
```
Comment (RonanJackson) [inline src/components/AddToMap.js:83]: We don't need the `classNames` parameters if we don't specify anything?

File: src/components/AddToMap.js [hunk 3]
```diff
@@ -52,36 +58,78 @@ function AddToMap() {
...
+          required
+          type="text"
+          placeholder="Enter longitude"
+          value={longitudeVal}
+          onChange={(e) => setLongitudeVal(e.target.value)}
+          disabled={!!disabledClass}
+          className={`${disabledClass} mt-4 shadow appearance-none border rounded w-full py-2 px-3 text-gray-600 focus:outline-none focus:shadow-outline`}
+        />
+      </div>
+      <div className="">
```
Comment (RonanJackson) [inline src/components/AddToMap.js:98]: We don't need the `classNames` parameters if we don't specify anything?

File: src/components/AddToMap.js [hunk 4]
```diff
@@ -20,19 +16,29 @@ const BUSINESS_UPDATE_MAP_INFO = gql`
...
-      title: title.value,
-      longitude: parseFloat(longitude.value),
-      latitude: parseFloat(latitude.value),
+      username: username,
+      title: titleVal,
+      longitude: parseFloat(longitudeVal),
+      latitude: parseFloat(latitudeVal),
     }
 
     if (!validAddToMapInputs(businessInfo)) {
```
[the remark this answers is not in the document]
Comment (RonanJackson) [reply src/components/AddToMap.js:44]: Also, should this function be extracted to the utils folder at all, I don't see it being reused, and it seems pretty specific to this react file?

File: src/components/Login.js [hunk 5]
```diff
@@ -99,31 +107,52 @@ export default function Login() {
...
+          value={otp}
+          onChange={(e) => setOtp(e.target.value)}
+          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 focus:outline-none focus:shadow-outline"
+        />
+        <p className="text-red-500 mt-2 text-xs italic">
+          {otp && !validAuthCode(otp) ? "Invalid auth code" : ""}
+        </p>
+        <button
+          type="submit"
+          disabled={!validAuthCode(otp)}
```
Comment (RonanJackson) [inline src/components/Login.js:126]: Similar to the above comment about ! in front of conditionals, could we change this to make it xxx===yyy?

File: src/components/Login.js [hunk 6]
```diff
@@ -99,31 +107,52 @@ export default function Login() {
...
+            <img
+              src={process.env.PUBLIC_URL + "/logo.png"}
+              alt="Bitcoin Beach logo"
+              className="m-auto filter invert"
+            />
+          </div>
+          <main className="p-6 sm:p-12 md:w-1/2">
+            <h1 className="block text-3xl font-extrabold text-gray-900">Welcome</h1>
+            <div className="py-4">
+              {!otpGenerated && <PhoneForm />}
```
Comment (RonanJackson) [inline src/components/Login.js:149]: Similar to the above comment about ! in front of conditionals, could we change this to make it xxx===yyy?

File: src/components/UserDetails.js [hunk 7]
```diff
@@ -92,10 +88,21 @@ const USER_UPDATE_LEVEL = gql`
   }
 `
 
+const emptyUserDetails = {
```
Comment (RonanJackson) [inline src/components/UserDetails.js:91]: Maybe rename this to: `defaultUserDetails`?

File: src/components/UserDetails.js [hunk 8]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
     updateUserLevel({ variables: { input: { uid: userDetails.id, level: "TWO" } } })
   }
 
+  const [searchValue, setSearchValue] = useState("")
+
+  const search = () => {
+    if (!searchValue) return
```
Comment (RonanJackson) [inline src/components/UserDetails.js:179]: Similar to above, can we have this not start with a `!`?

File: src/components/UserDetails.js [hunk 9]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
     updateUserLevel({ variables: { input: { uid: userDetails.id, level: "TWO" } } })
   }
 
+  const [searchValue, setSearchValue] = useState("")
```
Comment (RonanJackson) [inline src/components/UserDetails.js:176]: Should this be at the top of the function next to line 106?

File: src/components/UserDetails.js [hunk 10]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
...
+  }
+
+  const userDetailsUpdated = ({ username }) => {
+    getUserByUsername({ variables: { username } })
+  }
+
+  const loading = gettingUserByPhone || gettingUserByUsername
+  let detailClass = userDetails === emptyUserDetails || loading ? "filter blur-sm" : ""
+  detailClass = detailClass + (loading ? " animate-pulse" : "")
+  if (!userDetails) setUserDetails(emptyUserDetails)
```
Comment (RonanJackson) [inline src/components/UserDetails.js:200]: Starts with `!`, can we change?

File: src/components/UserDetails.js [hunk 11]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
...
+          <div className="col-span-2">
+            <p className="mb-4 font-semibold text-gray-600">Created At</p>
+            <p className={`text-gray-600 ${detailClass}`}>
+              {new Date(userDetails.createdAt * 1e3).toString()}
+            </p>
+          </div>
+        </div>
+        <div className="grid grid-cols-1 gap-4">
+          <div className="shadow p-6 min-w-0 rounded-lg shadow-xs overflow-hidden bg-white grid grid-cols-2 gap-4">
+            <div className="">
```
Comment (RonanJackson) [inline src/components/UserDetails.js:264]: We don't need `className` if it's empty?

File: src/components/UserDetails.js [hunk 12]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
...
+                  <button
+                    onClick={changeLevel}
+                    className="text-sm mx-4 bg-green-500 hover:bg-green-700 text-white font-bold p-2 border border-green-700 rounded disabled:opacity-50"
+                  >
+                    Upgrade
+                  </button>
+                )}
+              </p>
+            </div>
+            <div className="">
```
Comment (RonanJackson) [inline src/components/UserDetails.js:278]: We don't need `className` if it's empty?

File: src/components/UserDetails.js [hunk 13]
```diff
@@ -173,99 +173,149 @@ function UserDetails() {
...
+          <small className="animate-pulse font-thin text-sm"> (loading...)</small>
+        )}
+      </h1>
+      <div className="grid gap-6 mb-8 md:grid-cols-2 p-6">
+        <div className="shadow p-6 min-w-0 rounded-lg shadow-xs overflow-hidden bg-white grid grid-cols-2 gap-4">
+          <div className="">
+            <p className="mb-4 font-semibold text-gray-600">Phone</p>
+            <p className={`text-gray-600 ${detailClass}`}>{userDetails.phone}</p>
+          </div>
+          <div className="">
```
Comment (RonanJackson) [inline src/components/UserDetails.js:222]: We don't need `className` if it's empty?

**Comment by dolcalmi** [comment]
closed in favor of https://github.com/GaloyMoney/admin-panel/pull/13 
- Includes all Ronan's suggestions/fixes
- Includes transactions screens

**Pull request closed without merging**